Content.Server/Weapon/Ranged/GunSystem.Battery.cs

```diff
@@ -35,10 +35,16 @@ private void OnBatteryGetState(EntityUid uid, BatteryBarrelComponent component,
         args.State = new BatteryBarrelComponentState(
             component.FireRateSelector,
             count);
+
+        component.LastShotsLeftSent = count?.Item1;
     }
 
     private void OnCellSlotUpdated(EntityUid uid, BatteryBarrelComponent component, PowerCellChangedEvent args)
     {
+        // Update item-status information.
+        if (component.LastShotsLeftSent != component.ShotsLeft)
+            Dirty(component);
+
         UpdateBatteryAppearance(component);
     }
```

Comment:
Does appearancecomp not handle it?

Realistically most of this needs refactoring for gun prediction so it's going to be temporary anyway so eh.

Reply:
No, the item visual stuff goes through the component which implements `IItemStatus`, which doesn't use appearance data / know when appearance data gets updated.

But yeah a more elegant solution to this is probably to just subscribe to appearance events and update the UI then (rather than what it does ATM, where it updates on component state handling). Ill do that

Reply:
OK, now there is just a system listening for AppearanceChangeEvent and then updating the GUI